src/nodes/conv.rs

```diff
@@ -50,6 +51,12 @@ fn system_time_to_timespec(path: &Path, name: &str, time: &io::Result<SystemTime
     }
 }
 
+/// Converts a `time::Timespec` object into a `sys::time::TimeVal`.
+pub fn timespec_to_timeval(spec: Timespec) -> sys::time::TimeVal {
+    const S_TO_NS: i64 = 1000 * 1000 * 1000;
+    sys::time::TimeVal::nanoseconds(spec.sec * S_TO_NS + i64::from(spec.nsec))
```

Comment:
Why `::from`?

I recommend doing this by composition: `TimeVal::seconds(spec.sec) + TimeVal::nanoseconds(spec.nsec)`

Reply:
Nice. Done.

Before you ask, the `into()` is necessary because `spec.nsec` is an `i32` but the nanoseconds constructor expects an `i64`.